### PR TITLE
AC- 7597 Create mentor confirmation page

### PR DESCRIPTION
**Pull request by geofrocker**
### Changes introduced in: [AC- 7597](https://masschallenge.atlassian.net/browse/AC-7597) :
- Added graphql query
#### On test3
- Masquerade as any mentor e.g [this user](https://test3.masschallenge.org/admin/simpleuser/user/1032/masquerade/)
- Execute this graphql [query](https://test3api.masschallenge.org/graphql/#query=%7B%0A%20%20expertProfile%20%7B%0A%20%20%20%20mentorProgramRoleGrants%0A%20%20%7D%0A%7D%0A&variables=null&operationName=undefined)
- Notice that the endpoint returns the users mentor program role grants
#### On Localhost
- Masquerade as any mentor e.g [this user](http://localhost:8181/admin/simpleuser/user/1032/masquerade/)
- Execute this graphql [query](http://localhost:8000/graphql/#query=%7B%0A%20%20expertProfile%20%7B%0A%20%20%20%20mentorProgramRoleGrants%0A%20%20%7D%0A%7D%0A&variables=null&operationName=undefined)
- Notice that the endpoint returns the users mentor program role grants

### CODE REVIEW

**Comment by iankigen** [comment]
Doing QA

File: web/impact/impact/graphql/types/base_user_profile_type.py [hunk 1]
```diff
@@ -31,3 +37,39 @@ def resolve_program_roles(self, info, **kwargs):
             startup_roles_of_interest += [StartupRole.ENTRANT]
         return get_user_program_and_startup_roles(
             self.user, user_roles_of_interest, startup_roles_of_interest)
+
+    def resolve_mentor_program_role_grants(self, info, **kwargs):
+        """
+        Returns the mentor program grants for a user
+        """
+        roles = ["Mentor", "Desired Mentor", "Deferred Mentor"]
```
Comment (iankigen) [inline web/impact/impact/graphql/types/base_user_profile_type.py]: can we use constants here ie `UserRole.MENTOR`

File: web/impact/impact/v1/views/mentor_program_office_hour_list_view.py [hunk 2]
```diff
@@ -45,6 +47,14 @@ def _filter_by_user_open_hours(self, qs):
         qs = qs.filter(program_id__in=program_ids, finalist__isnull=True)
         return qs
 
+    def _filter_by_user_all_hours(self, qs):
```
Comment (iankigen) [inline web/impact/impact/v1/views/mentor_program_office_hour_list_view.py]: I'm not seeing how changes in this file are related to this ticket

File: web/impact/impact/tests/test_graphql.py [hunk 3]
```diff
@@ -97,11 +99,43 @@ def test_query_with_expert(self):
                     'availableOfficeHours': False,
                     'officeHoursUrl': None,
                     'programInterests': [],
+                    'mentorProgramRoleGrants': {'results': []}
                 }
             }
         }
         self._assert_response_equals_json(query, expected_json)
 
+    def test_mentor_program_role_grants(self):
```
Comment (iankigen) [inline web/impact/impact/tests/test_graphql.py:116]: can we include another test to confirm that `mentorProgramRoleGrants` will only return program role grants where user role is ["Mentor", "Desired Mentor", "Deferred Mentor"] even if the user has other user roles

File: web/impact/impact/v1/helpers/mentor_program_office_hour_helper.py [hunk 4]
```diff
@@ -19,9 +19,12 @@
     "end_date_time": OPTIONAL_STRING_FIELD,
     "mentor_id": OPTIONAL_INTEGER_FIELD,
     "mentor_name": OPTIONAL_STRING_FIELD,
+    "mentor_title": OPTIONAL_STRING_FIELD,
```
Comment (iankigen) [inline web/impact/impact/v1/helpers/mentor_program_office_hour_helper.py]: I'm not seeing how changes in this file are related to this ticket

**Review by iankigen — CHANGES_REQUESTED** [review]

File: web/impact/impact/v1/helpers/mentor_program_office_hour_helper.py (hunk 4 above)
Comment (geofrocker) [reply web/impact/impact/v1/helpers/mentor_program_office_hour_helper.py]: Nice catch

File: web/impact/impact/v1/views/mentor_program_office_hour_list_view.py (hunk 2 above)
Comment (geofrocker) [reply web/impact/impact/v1/views/mentor_program_office_hour_list_view.py]: Some other ticket's work. Removing it ...thanks

**Review by iankigen — APPROVED** [review]
QA/Code looks good.

Confirmed:
- Graphql field returning mentorProgramRoleGrants of user role type [mentor, desired mentor and deferred mentor] was added

**Review by freshprincekla — APPROVED** [review]
Code and QA look good.
mentorProgramRoleGrants for mentor, desired mentor and deferred mentor were return in a GraphQL field.

**Pull request merged**